The test is failing with
```
2026-06-09T23:03:40.506Z [ERROR] sdk.helper_resource: Unexpected error: test_working_directory=/tmp/plugintest1153407796 test_name=TestAccArtifactRegistryRepository_remoteWithAuthUpdate test_terraform_path=/usr/bin/terraform
  error=
  | After applying this test step, the non-refresh plan was not empty.
  | stdout:
  | 
  | 
  | Terraform used the selected providers to generate the following execution
  | plan. Resource actions are indicated with the following symbols:
  |   ~ update in-place
  | 
  | Terraform will perform the following actions:
  | 
  |   # google_artifact_registry_repository.my-repo will be updated in-place
  |   ~ resource "google_artifact_registry_repository" "my-repo" {
  |         id                     = "projects/ci-test-project-188019/locations/us-central1/repositories/tf-test-my-repository-pqp8wpyzfm"
  |         name                   = "tf-test-my-repository-pqp8wpyzfm"
  |         # (15 unchanged attributes hidden)
  | 
  |       ~ remote_repository_config {
  |             # (2 unchanged attributes hidden)
  | 
  |           ~ upstream_credentials {
  |               ~ username_password_credentials {
  |                   ~ password_secret_version = "projects/1067888929963/secrets/tf-test-example-secret-pqp8wpyzfm/versions/1" -> "projects/1067888929963/secrets/tf-test-example-secret-pqp8wpyzfm/versions/2"
  |                   ~ username                = "tf-test-username" -> "tf-test-username-new"
  |                 }
  |             }
  | 
  |             # (1 unchanged block hidden)
  |         }
  | 
  |         # (1 unchanged block hidden)
  |     }
  | 
  | Plan: 0 to add, 1 to change, 0 to destroy.
   test_step_number=3
   ```
   
   However, I don't see a `PATCH` request anywhere in the log, so I'm not sure why it's expecting the updated values.

--- a/mmv1/products/artifactregistry/Repository.yaml
+++ b/mmv1/products/artifactregistry/Repository.yaml
@@ -761,7 +761,6 @@ properties:
         type: NestedObject
         description: |-
           The credentials used to access the remote repository.
-        is_missing_in_cai: true
         properties:
           - name: 'usernamePasswordCredentials'
             type: NestedObject

--- a/mmv1/products/artifactregistry/Repository.yaml
+++ b/mmv1/products/artifactregistry/Repository.yaml
@@ -506,7 +506,6 @@ properties:
     type: NestedObject
     description: |-
       Configuration specific for a Remote Repository.
-    immutable: true
     conflicts:
       - virtual_repository_config
     properties:

--- a/mmv1/products/artifactregistry/Repository.yaml
+++ b/mmv1/products/artifactregistry/Repository.yaml
@@ -53,6 +53,7 @@ iam_policy:
 custom_code:
   constants: 'templates/terraform/constants/artifact_registry_repository.go.tmpl'
   encoder: 'templates/terraform/encoders/location_from_region.go.tmpl'
+  pre_update: 'templates/terraform/update_encoder/artifact_registry_repository_update.go.tmpl'
 sweeper:
   url_substitutions:
     - region: "us-central1"
@@ -506,6 +507,7 @@ properties:
     type: NestedObject
     description: |-
       Configuration specific for a Remote Repository.
+    immutable: true
     conflicts:
       - virtual_repository_config
     properties:

--- a/mmv1/templates/terraform/update_encoder/artifact_registry_repository_update.go.tmpl
+++ b/mmv1/templates/terraform/update_encoder/artifact_registry_repository_update.go.tmpl
@@ -1,0 +1,20 @@
+
+remoteRepositoryConfigProp, err := expandArtifactRegistryRepositoryRemoteRepositoryConfig(d.Get("remote_repository_config"), d, config)
+	if err != nil {
+		return err
+	} else if v, ok := d.GetOkExists("remote_repository_config"); !tpgresource.IsEmptyValue(reflect.ValueOf(remoteRepositoryConfigProp)) && (ok || !reflect.DeepEqual(v, remoteRepositoryConfigProp)) {
+		obj["remoteRepositoryConfig"] = remoteRepositoryConfigProp
+	}
+
+if d.HasChange("remote_repository_config.0.upstream_credentials.0.username_password_credentials.0.username") {
+  updateMask = append(updateMask, "remoteRepositoryConfig.upstreamCredentials.usernamePasswordCredentials.username")
+}
+
+if d.HasChange("remote_repository_config.0.upstream_credentials.0.username_password_credentials.0.password_secret_version") {
+  updateMask = append(updateMask, "remoteRepositoryConfig.upstreamCredentials.usernamePasswordCredentials.passwordSecretVersion")
+}
+
+url, err = transport_tpg.AddQueryParams(url, map[string]string{"updateMask": strings.Join(updateMask, ",")})
+if err != nil {
+  return err
+}

